AppCenter/AppCenter/Internals/Util/MSEncrypter.m

```diff
@@ -117,52 +65,150 @@ - (NSString *_Nullable)decryptString:(NSString *)string {
 }
 
 - (NSData *_Nullable)decryptData:(NSData *)data {
-  [self createCryptorContexts:kMSEncryptionKeyTag];
-  if (!self.key) {
-    MSLogError([MSAppCenter logTag], @"Could not perform decryption. Encryption key is missing.");
-    return nil;
+
+  // Separate cipher prefix from cipher.
+  NSRange dataRange = NSMakeRange(0, [data length]);
+  NSData *separatorAsData = [kMSEncryptionMetadataSeparator dataUsingEncoding:NSUTF8StringEncoding];
+  size_t metadataLocation = [data rangeOfData:separatorAsData options:0 range:dataRange].location;
+  NSString *metadata;
+  if (metadataLocation != NSNotFound) {
+    NSData *subdata = [data subdataWithRange:NSMakeRange(0, metadataLocation)];
+    metadata = [[NSString alloc] initWithData:subdata encoding:NSUTF8StringEncoding];
   }
-  NSData *result = nil;
-  size_t cipherBufferSize = CCCryptorGetOutputLength(self.decryptorObject, data.length, true);
-  uint8_t *cipherBuffer = malloc(cipherBufferSize);
-  size_t numBytesDecrypted = 0;
-  CCCryptorStatus status = CCCrypt(kCCDecrypt, kMSEncryptionAlgorithm, kCCOptionPKCS7Padding, [self.key bytes], kMSCipherKeySize, nil,
-                                   [data bytes], data.length, cipherBuffer, cipherBufferSize, &numBytesDecrypted);
-  if (status != kCCSuccess) {
-    MSLogError([MSAppCenter logTag], @"Error performing decryption with CCCryptorStatus: %d.", status);
+  NSData *key;
+  NSData *initializationVector;
+  NSData *cipherText;
+  if (!metadata) {
```

Comment:
Why don't you swap `if` block and `else` block by reversing the condition? Don't need NOT operation.